test/hotspot/jtreg/compiler/vectorapi/reshape/utils/VectorReshapeHelper.java

```diff
@@ -315,58 +301,28 @@ public static void runMainHelper(Class<?> testClass, Stream<VectorSpeciesPair> t
         }
     }
 
-    public static byte getByte(byte[] array, int index) {
-        return (byte)BYTE_ACCESS.get(array, index * Byte.BYTES);
-    }
-
-    public static short getShort(byte[] array, int index) {
-        return (short)SHORT_ACCESS.get(array, index * Short.BYTES);
-    }
-
-    public static int getInt(byte[] array, int index) {
-        return (int)INT_ACCESS.get(array, index * Integer.BYTES);
-    }
-
-    public static long getLong(byte[] array, int index) {
-        return (long)LONG_ACCESS.get(array, index * Long.BYTES);
-    }
-
-    public static float getFloat(byte[] array, int index) {
-        return (float)FLOAT_ACCESS.get(array, index * Float.BYTES);
-    }
-
-    public static double getDouble(byte[] array, int index) {
-        return (double)DOUBLE_ACCESS.get(array, index * Double.BYTES);
-    }
-
-    public static void setByte(byte[] array, int index, byte value) {
-        BYTE_ACCESS.set(array, index * Byte.BYTES, value);
-    }
-
-    public static void setShort(byte[] array, int index, short value) {
-        SHORT_ACCESS.set(array, index * Short.BYTES, value);
-    }
-
-    public static void setInt(byte[] array, int index, int value) {
-        INT_ACCESS.set(array, index * Integer.BYTES, value);
-    }
-
-    public static void setLong(byte[] array, int index, long value) {
-        LONG_ACCESS.set(array, index * Long.BYTES, value);
-    }
-
-    public static void setFloat(byte[] array, int index, float value) {
-        FLOAT_ACCESS.set(array, index * Float.BYTES, value);
+    @ForceInline
+    private static <T> Vector<T> readVector(VectorSpecies<T> isp, Object input) {
+        return isp.fromArray(input, 0);
     }
 
-    public static void setDouble(byte[] array, int index, double value) {
-        DOUBLE_ACCESS.set(array, index * Double.BYTES, value);
+    @ForceInline
+    private static <U> void writeVector(VectorSpecies<U> osp, Vector<U> vector, Object output) {
+        var otype = osp.elementType();
```

Comment:
This code suggests we are missing a generic array store method to compliment the generic array load method.